lib/constructSignatureString.test.js

```diff
@@ -2,15 +2,18 @@
 
 const { test } = require('tap')
 const constructSignatureString = require('./constructSignatureString')
+const getDigest = require('./getDigest')
 
 const defaultOptions = {
   sharedSecret: null,
   getAlgorithm: () => {
     throw new Error('No getAlgorithm function provided.')
   },
-  getDigest: () => {
-    throw new Error('No getDigest function provided.')
-  }
+  getHMACEncoding: () => {
+    throw new Error('No getHMACEncoding function provided.')
+  },
```

Comment:
These will never be not be defined right? We have default values for them when implementing the plugin. Unless someone explicitly voids them by passing an options object like: 

```js
{ getAlgorithm: null }
```

This test should actually probably check the type of the passed option (we want to enforce passing a function). But I think we should test that at the plugin implementation level, not here.

In the meantime I think we can remove these test cases

Reply:
Currently, we have the defaults defined as functions that return errors unless they are overloaded with proper functions that return algorithm/encoding strings during registration. See plugin.js.

These tests mimic that default setup but I think you're right that this kind of testing needs to happen at a higher level. I'll remove these test from this test file.